nemo/collections/tts/data/vocoder_dataset.py

```diff
@@ -175,28 +238,294 @@ def __getitem__(self, index):
-            "audio_lens": batch_audio_len,
+    def __init__(
+        self,
+        *,
+        dataset_meta: Dict,
+        sample_rate: int,
+        n_samples: Optional[int] = None,
+        shuffle_n: int = 0,
+        min_duration: float = 0.1,
+        max_duration: Optional[float] = None,
+        trunc_duration: Optional[float] = None,
+        feature_processors: Optional[Dict[str, FeatureProcessor]] = None,
+        num_audio_retries: int = 5,
+        shard_strategy: str = "scatter",
+        global_rank: int = 0,
+        world_size: int = 2,
+    ):
+        super().__init__()
+        self.sample_rate = sample_rate
+        self.n_samples = n_samples
+        self.num_audio_retries = num_audio_retries
+        self.load_precomputed_mel = False
+
+        if trunc_duration:
+            self.trunc_samples = int(trunc_duration * self.sample_rate)
+        else:
+            self.trunc_samples = None
+
+        if feature_processors:
+            logging.info(f"Found feature processors {feature_processors.keys()}")
+            self.feature_processors = list(feature_processors.values())
+        else:
+            self.feature_processors = []
+
+        self.data_samples = []
+        self.sample_weights = []
+        self.audio_tar_filepaths = []
+        for dataset_name, dataset_info in dataset_meta.items():
+            audio_tar_filepaths = dataset_info['audio_tar_filepaths']
+            self.audio_tar_filepaths += audio_tar_filepaths
+            dataset = DatasetMeta(**dataset_info)
+            samples, weights = VocoderDataset._preprocess_manifest(
+                dataset_name=dataset_name, dataset=dataset, min_duration=min_duration, max_duration=max_duration,
+            )
+            self.data_samples += samples
+            self.sample_weights += weights
+        self.mapping = {}
+        for idx, sample in enumerate(self.data_samples):
+            file_id = os.path.splitext(os.path.basename(sample.manifest_entry["audio_filepath"]))[0]
+            if file_id not in self.mapping:
+                self.mapping[file_id] = sample
+            else:
+                raise ValueError(
+                    f"Duplicate file_id {file_id} found in manifest {sample.manifest_entry['audio_filepath']}"
+                )
+
+        logging.info(f"world size: {world_size}")
+        audio_tar_filepaths = expand_sharded_filepaths(
+            sharded_filepaths=audio_tar_filepaths,
+            global_rank=global_rank,
+            world_size=world_size,
+            shard_strategy=shard_strategy,
+        )
+
+        self._dataset = wd.WebDataset(audio_tar_filepaths, nodesplitter=None)
+
+        if shuffle_n > 0:
+            self._dataset = self._dataset.shuffle(shuffle_n)
+        else:
+            logging.info("WebDataset will not shuffle data. Consider setting shuffle_n > 0.")
+
+        self._dataset = (
+            self._dataset.rename(audio=VALID_FILE_FORMATS, key='__key__')
+            .to_tuple('audio', 'key')
```

Comment:
Nitpick: it is not very important for audio only dataset, but code would be more readable and extensible using original dictionary format instead of converting to tuple.